pyramid_favicon/favicon.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+"""Serve the favicon.ico at the application root."""
+
+from pyramid.response import FileResponse
+from pyramid.security import NO_PERMISSION_REQUIRED
+from pyramid.view import view_config
+
+import os
+
+
+@view_config(
+    route_name='favicon',
+    permission=NO_PERMISSION_REQUIRED,
+)
+def favicon(request):
+    
+    here = os.path.dirname(__file__)
+    
+    settings = request.registry.settings
+    favicon_path = settings.get('favicon_path', None)
+    if favicon_path:
+        icon = os.path.join(favicon_path, 'favicon.ico')
```

Comment:
For finding view templates and such pyramid supports giving a "dotted name" instead of a path and then pyramid resolves it to a full path. We should support the same here.

So something like "mypackage:images/favicon.ico" should work.